source/state_representation/include/state_representation/Space/Cartesian/CartesianState.hpp

```diff
@@ -5,476 +5,715 @@
+   */
+  const CartesianState operator-(const CartesianState& state) const;
+
+  /**
+   * @brief compute the inverse of the current CartesianState
+   * @return the inverse corresponding to b_S_f (assuming this is f_S_b) 
+   */
+  const CartesianState inverse() const;
+
+  /**
+   * @brief Compute the distance between two states as the sum of distances between each features
+   * @param state the second state
+   * @param state_variable_type name of the variable from the CartesianStateVariable structure to apply
+   * the distance on. Default ALL for full distance across all dimensions
+   * @return dist the distance value as a double
+   */
+  double dist(const CartesianState& state, const CartesianStateVariable& state_variable_type = CartesianStateVariable::ALL) const;
+
+  /**
+   * @brief Overload the ostream operator for printing
+   * @param os the ostream to happend the string representing the state to
+   * @param state the state to print
+   * @return the appended ostream
+   */
+  friend std::ostream& operator<<(std::ostream& os, const CartesianState& state);
+
+  /**
+   * @brief Overload the * operator with a scalar
+   * @param lambda the scalar to multiply with
+   * @return the CartesianState provided multiply by lambda
+   */
+  friend const CartesianState operator*(double lambda, const CartesianState& state);
+
+  /**
+   * @brief compute the distance between two CartesianStates
+   * @param s1 the first CartesianState 
+   * @param s2 the second CartesianState
+   * @param type of the distance between position, orientation, linear_velocity, etc...
+   * default all for full distance across all dimensions
+   * @return the distance beteen the two states
+   */
+  friend double dist(const CartesianState& s1, const CartesianState& s2, const std::string& distance_type);
+
+  /**
+   * @brief Return the state as a std vector of floats
+   * @return std::vector<float> the state vector as a std vector
+   */
+  virtual const std::vector<double> to_std_vector() const;
+
+  /**
+   * @brief Set the value from a std vector
+   * @param value the value as a std vector
+   */
+  virtual void from_std_vector(const std::vector<double>& value);
+};
+
+inline CartesianState& CartesianState::operator=(const CartesianState& state) {
+  SpatialState::operator=(state);
+  this->set_pose(state.get_pose());
+  this->set_twist(state.get_twist());
+  this->set_accelerations(state.get_accelerations());
+  this->set_wrench(state.get_wrench());
+  return (*this);
+}
+
+inline const Eigen::Vector3d& CartesianState::get_position() const {
+  return this->position;
+}
+
+inline const Eigen::Quaterniond& CartesianState::get_orientation() const {
+  return this->orientation;
+}
+
+inline const Eigen::Vector4d CartesianState::get_orientation_coefficients() const {
+  return Eigen::Vector4d(this->get_orientation().w(), this->get_orientation().x(), this->get_orientation().y(), this->get_orientation().z());
```

Comment:
I voluntarily change it here to be consistent in all the functions. I don't like Eigen convention for this and I think it is important that we are consistent. But this can be open indeed